analyses/doublet-detection/run_doublet-detection-scpca.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+# This script runs doublet detection over ScPCA data for a given project
+# Usage: ./run_doublet-detection-scpca.sh {scpca project id}
+
+set -euo pipefail
+
+
+# Ensure script is being run from its directory
+#MODULE_DIR=$(dirname "${BASH_SOURCE[0]}")
+#cd ${MODULE_DIR}
+
+PROJECT_ID=$1
+
+# Define directories
+DATA_DIR="../../data/current"
+RESULTS_DIR="results/scpca-results/${PROJECT_ID}"
+mkdir -p ${RESULTS_DIR}
+
+# Detect doublets on each processed SCE file in each sample directory
+for SAMPLE_DIR in ${DATA_DIR}/${PROJECT_ID}/SCPCS*; do
+    SAMPLE_ID=$(basename $SAMPLE_DIR)
+    echo "Processing ${SAMPLE_ID}..."
+
+    SAMPLE_RESULTS_DIR=${RESULTS_DIR}/${SAMPLE_ID}
+    mkdir -p ${SAMPLE_RESULTS_DIR}
+
+    for SCE_FILE in $(ls ${SAMPLE_DIR}/*processed.rds); do
```

Comment:
It is better not to use `ls` in for loops; just the glob is sufficient
```suggestion
    for SCE_FILE in ${SAMPLE_DIR}/*_processed.rds; do
```